crates/bevy_tasks/src/task_pool.rs

```diff
@@ -415,4 +441,31 @@ mod tests {
         assert!(!thread_check_failed.load(Ordering::Acquire));
         assert_eq!(count.load(Ordering::Acquire), 200);
     }
+
+    #[test]
+    fn test_spawn_pollable() {
+        let transform_fn = |i| i + 1;
+
+        let pool = TaskPool::new();
+        let nums: Range<u8> = 0..10;
+
+        let pollable_tasks = nums
+            .clone()
+            .into_iter()
+            .map(|i| pool.spawn_pollable(async move { transform_fn(i) }))
+            .collect::<Vec<_>>();
+
+        for _ in 0..100 {
+            for (pollable_task, number) in pollable_tasks.iter().zip(nums.clone()) {
+                match pollable_task.poll() {
+                    None => continue,
+                    Some(actual) => assert_eq!(transform_fn(number), actual),
+                }
+                return;
+            }
+
+            std::thread::sleep(Duration::from_secs_f32(1. / 100.));
```

Comment:
Sleeping in tests is very much not ideal. I think we would check the output from the relevant channel from each task

I.e. write this test relatviely `async`, using e.g. `TaskPoll::scope`

Reply:
I don't quite get how I should use `TaskPool::scope` here. I want to test that a `PollableTask` will return something. Should I wrap those pollable tasks in a scope to do the waiting there?